src/main/java/com/google/sps/servlets/ChatroomServlet.java

```diff
@@ -36,37 +36,50 @@
 import com.google.sps.servlets.Message;
 import com.google.sps.servlets.Post;
 import com.google.sps.servlets.Chatroom;
+import com.pusher.rest.Pusher;
 
 /** Servlet that holds the chatrooms active on this WebApp */
 @WebServlet("/chatroom")
 public class ChatroomServlet extends HttpServlet {
 
-
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
         String userID = request.getParameter("userId");
         String recipientID = request.getParameter("recipientId"); // the user whose chat was clicked
-        String chatroomID = "";
+        String chatroomID = "null";
 
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         Query chatroomQuery = new Query("chatroom");
         PreparedQuery chatrooms = datastore.prepare(chatroomQuery);
+
         for (Entity chatroom : chatrooms.asIterable()) {
             String chatroomId = (String) chatroom.getProperty("chatroomId");
-            ArrayList<String> users = (ArrayList<String>) chatroom.getProperty("users");
+            String user1 = (String) chatroom.getProperty("user1");
+            String user2 = (String) chatroom.getProperty("user2");
 
-            Chatroom currChatRoom = new Chatroom(chatroomId, users.get(0), users.get(1));
+            Chatroom currChatRoom = new Chatroom(chatroomId, user1, user2);
+            List<String> usersList = currChatRoom.getUsers();
 
-            if (currChatRoom.users.contains(userID) && currChatRoom.users.contains(recipientID)) {
-                    chatroomID = currChatRoom.chatroomId;
+            if ((usersList.contains(userID)) && (usersList.contains(recipientID))) {
+                    chatroomID = currChatRoom.getId();
                     break;
             }
         }
 
+        // Check to see if chatroom is empty and make new chatroom
+        if (chatroomID == "null") {
+            Entity newChatroom = new Entity("chatroom");
+            chatroomID = UUID.randomUUID().toString();
```

Comment:
random q: is there any chance whatsoever that UUID may create a number similar to one of the existing chatroomIDs?